### ISSUE TITLE
Cookie Consent doesnt save and doesnt disappear on Reload

### ISSUE THREAD

**Issue by TryAnixx** [orestbida/cookieconsent#322]
Hello,

i choose something on my cookie consent but on reload my popup gets triggerd again.
How can i save the values?

HTML:
```
<!DOCTYPE html>
<html>
<head>
    <title>Hello World</title>
    <link href="https://cdn.jsdelivr.net/gh/orestbida/cookieconsent@v2.4.5/dist/cookieconsent.css" rel="stylesheet">
</head>
<body>
    <h1>Hello World</h1>
    <script defer src="https://cdn.jsdelivr.net/gh/orestbida/cookieconsent@v2.8.0/dist/cookieconsent.js"></script>
    <script defer src="cookieconsent-init.js"></script>   
</body>
</html>
```

Cookie Consent Init:
```
// obtain plugin
var cc = initCookieConsent();

// run plugin with your configuration
cc.run({
    current_lang: 'en',
    autoclear_cookies: true,                   // default: false
    page_scripts: true,                        // default: false

    // mode: 'opt-in'                          // default: 'opt-in'; value: 'opt-in' or 'opt-out'
    // delay: 0,                               // default: 0
    // auto_language: null                     // default: null; could also be 'browser' or 'document'
    // autorun: true,                          // default: true
    // force_consent: false,                   // default: false
    // hide_from_bots: false,                  // default: false
    // remove_cookie_tables: false             // default: false
    // cookie_name: 'cc_cookie',               // default: 'cc_cookie'
    // cookie_expiration: 182,                 // default: 182 (days)
    // cookie_necessary_only_expiration: 182   // default: disabled
    // cookie_domain: location.hostname,       // default: current domain
    // cookie_path: '/',                       // default: root
    // cookie_same_site: 'Lax',                // default: 'Lax'
    // use_rfc_cookie: false,                  // default: false
     //revision: 0,                            // default: 0

    onFirstAction: function(user_preferences, cookie){
        // callback triggered only once
    },

    onAccept: function (cookie) {
        // ...
    },

    onChange: function (cookie, changed_preferences) {
        // ...
    },

    languages: {
        'en': {
            consent_modal: {
                title: 'We use cookies!',
                description: 'Hi, this website uses essential cookies to ensure its proper operation and tracking cookies to understand how you interact with it. The latter will be set only after consent. <button type="button" data-cc="c-settings" class="cc-link">Let me choose</button>',
                primary_btn: {
                    text: 'Accept all',
                    role: 'accept_all'              // 'accept_selected' or 'accept_all'
                },
                secondary_btn: {
                    text: 'Reject all',
                    role: 'accept_necessary'        // 'settings' or 'accept_necessary'
                }
            },
            settings_modal: {
                title: 'Cookie preferences',
                save_settings_btn: 'Save settings',
                accept_all_btn: 'Accept all',
                reject_all_btn: 'Reject all',
                close_btn_label: 'Close',
                cookie_table_headers: [
                    {col1: 'Name'},
                    {col2: 'Domain'},
                    {col3: 'Expiration'},
                    {col4: 'Description'}
                ],
                blocks: [
                    {
                        title: 'Cookie usage 📢',
                        description: 'I use cookies to ensure the basic functionalities of the website and to enhance your online experience. You can choose for each category to opt-in/out whenever you want. For more details relative to cookies and other sensitive data, please read the full <a href="#" class="cc-link">privacy policy</a>.'
                    }, {
                        title: 'Strictly necessary cookies',
                        description: 'These cookies are essential for the proper functioning of my website. Without these cookies, the website would not work properly',
                        toggle: {
                            value: 'necessary',
                            enabled: true,
                            readonly: true          // cookie categories with readonly=true are all treated as "necessary cookies"
                        }
                    }, {
                        title: 'Performance and Analytics cookies',
                        description: 'These cookies allow the website to remember the choices you have made in the past',
                        toggle: {
                            value: 'analytics',     // your cookie category
                            enabled: false,
                            readonly: false
                        },
                        cookie_table: [             // list of all expected cookies
                            {
                                col1: '^_ga',       // match all cookies starting with "_ga"
                                col2: 'google.com',
                                col3: '2 years',
                                col4: 'description ...',
                                is_regex: true
                            },
                            {
                                col1: '_gid',
                                col2: 'google.com',
                                col3: '1 day',
                                col4: 'description ...',
                            }
                        ]
                    }, {
                        title: 'Advertisement and Targeting cookies',
                        description: 'These cookies collect information about how you use the website, which pages you visited and which links you clicked on. All of the data is anonymized and cannot be used to identify you',
                        toggle: {
                            value: 'targeting',
                            enabled: false,
                            readonly: false
                        }
                    }, {
                        title: 'More information',
                        description: 'For any queries in relation to our policy on cookies and your choices, please <a class="cc-link" href="#yourcontactpage">contact us</a>.',
                    }
                ]
            }
        }
    }
});
```

**Comment by orestbida** [comment]
Have you set-up a web server? Check the url in the browser, and make sure it begins with either http or https.

**Comment by BerkeAras** [comment]
Same issue. @orestbida 
Lastest "next"-Version.
I have a subdomain (shop.***.de), it always appears on every reload.

**Comment by orestbida** [comment]
@BerkeAras can you provide the configuration or the live address?

**Comment by BerkeAras** [comment]
@orestbida Sure:
Background-Information: I have one Site (React), but 2 different Domains for it.
Primary-Domain (works fine): https://bestellen.deliverone.de/
Secondary-Domain (shows up on every reload): https://shop.87616mod.de/

React-Code:
```jsx
CookieConsent.run({
  root: 'body',
  autoShow: true,
  disablePageInteraction: true,
  mode: 'opt-in',
  revision: 0,
  domain: window.location.hostname,
  cookie: {
    name: 'cc_cookie',
    domain: window.location.hostname,
    sameSite: 'None'
  },
  guiOptions: {
    consentModal: {
      layout: "box wide",
      position: "middle center",
      equalWeightButtons: true,
      flipButtons: false
    },
    preferencesModal: {
      layout: "box",
      position: "right",
      equalWeightButtons: true,
      flipButtons: false
    }
  },
  categories: {
    necessary: {
      enabled: true,
      readOnly: true
    },
    analytics: {
      autoClear: {
        cookies: [
          {
            name: /^_ga/,
          },
          {
            name: '_gid',
          },
          {
            name: /^_ga/,
          },
          {
            name: 'clid',
          },
          {
            name: /^_lr/,
          },
          {
            name: /^_pc_vis/,
          },
        ]
      },
      services: {
        ga: {
          label: 'Google Analytics',
          onAccept: () => {
            CookieConsent.loadScript(`https://www.googletagmanager.com/gtag/js?id=G-B5S0J7WC8B`, () => {}, [
              {name: 'async', value: ''}
            ]);
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', 'G-B5S0J7WC8B');
          },
          onReject: () => {}
        },
        logrocket: {
          label: 'LogRocket',
          onAccept: () => {
            LogRocket.init('oknhkd/deliverone');
          },
          onReject: () => {}
        }
      }
    },
    marketing: {
      autoClear: {
        cookies: [
          {
            name: /^_cl/
          },
          {
            name: /^CLID/
          },
          {
            name: /^MUID/
          },
          {
            name: /^SRM_B/
          },
          {
            name: /^MR/
          },
          {
            name: /^SM/
          },
          {
            name: /^ANONCHK/
          },
          {
            name: /^_clck/
          }
        ]
      },
      services: {
        clid: {
          label: 'Microsoft Clarity',
          onAccept: () => {
            (function(c,l,a,r,i,t,y){
              c[a]=c[a]||function(){(c[a].q=c[a].q||[]).push(arguments)};
              t=l.createElement(r);t.async=1;t.src="https://www.clarity.ms/tag/"+i;
              y=l.getElementsByTagName(r)[0];y.parentNode.insertBefore(t,y);
            })(window, document, "clarity", "script", "ep3bj834mq");
          },
          onReject: () => {}
        }
      }
    }
  },
  language: {
    default: "de",
    autoDetect: "browser",
    translations: {
      de: {
        consentModal: {
          title: "🍪 Wir verwenden Cookies!",
          description: 'Hallo, diese Website verwendet essentielle Cookies, um ihren ordnungsgemäßen Betrieb zu gewährleisten, und Tracking-Cookies, um zu verstehen, wie Du mit ihr interagierst. Letztere werden nur nach Zustimmung gesetzt.',
          acceptAllBtn: "Alle akzeptieren",
          acceptNecessaryBtn: "Alle ablehnen",
          showPreferencesBtn: "Einstellungen verwalten",
          footer: "<a href=\"https://deliverone.de/datenschutz\" target=\"_blank\">Datenschutz</a>\n<a href=\"https://deliverone.de/agb\" target=\"_blank\">Allgemeine Geschäftsbedingungen</a>"
        },
        preferencesModal: {
          title: "Präferenzen für die Zustimmung",
          acceptAllBtn: "Alle akzeptieren",
          acceptNecessaryBtn: "Alle ablehnen",
          savePreferencesBtn: "Einstellungen speichern",
          closeIconLabel: "Modal schließen",
          serviceCounterLabel: "Dienstleistungen",
          sections: [
            {
              title: 'Verwendung von Cookies 📢',
              description: 'Wir verwenden Cookies, um die grundlegenden Funktionen der Website zu gewährleisten und um dein Online-Erlebnis zu verbessern. Du kannst für jede Kategorie wählen, ob du dich an- oder abmelden möchtest. Weitere Informationen zu Cookies und anderen sensiblen Daten findest du in der <a href="https://deliverone.de/rechtliche-informationen/datenschutz" class="cc-link">Datenschutzerklärung</a>.'
            },
            {
              title: "Notwendige Cookies <span class=\"pm__badge\">Immer Aktiviert</span>",
              description: "Diese Cookies sind für das ordnungsgemäße Funktionieren der Website unerlässlich. Ohne diese Cookies würde die Website nicht richtig funktionieren",
              linkedCategory: "necessary",
              cookieTable: {
                headers: {
                  name: "Name",
                  domain: "Service",
                  description: "Zweck",
                  expiration: "Ablaufdatum"
                },
                body: [
                  {
                    name: "cc_cookie",
                    domain: window.location.hostname,
                    expiration: "6 Monate",
                    description: "Speichert Informationen zu ausgewählten Cookie-Präferenzen."
                  },
                  {
                    name: "PHPSESSID",
                    domain: window.location.hostname,
                    expiration: "Session",
                    description: "Behält die Zustände des Benutzers bei allen Seitenanfragen bei."
                  },
                  {
                    name: "ts",
                    domain: "paypal.com",
                    expiration: "3 Jahre",
                    description: "PayPal verwendet Cookies, die für die ordnungsgemäße Funktionsweise (Zahlungsvorgang) der Website erforderlich sind."
                  },
                  {
                    name: "ts_c",
                    domain: "paypal.com",
                    expiration: "3 Jahre",
                    description: "PayPal verwendet Cookies, die für die ordnungsgemäße Funktionsweise (Zahlungsvorgang) der Website erforderlich sind."
                  },
                  {
                    name: "__stripe_mid",
                    domain: window.location.hostname,
                    expiration: "1 Jahr",
                    description: "Stripe verwendet Cookies, die für die ordnungsgemäße Funktionsweise (Zahlungsvorgang) der Website erforderlich sind."
                  },
                  {
                    name: "__stripe_sid",
                    domain: window.location.hostname,
                    expiration: "Session",
                    description: "Stripe verwendet Cookies, die für die ordnungsgemäße Funktionsweise (Zahlungsvorgang) der Website erforderlich sind."
                  }
                ]
              }
            },
            {
              title: "Leistungs- und Analyse-Cookies",
              description: "Diese Cookies ermöglichen es der Website, sich an die Entscheidungen zu erinnern, die du in der Vergangenheit getroffen hast",
              linkedCategory: "analytics",
              cookieTable: {
                headers: {
                  name: "Name",
                  domain: "Service",
                  description: "Zweck",
                  expiration: "Ablaufdatum"
                },
                body: [
                  {
                    name: "^_ga",
                    domain: "google.com",
                    expiration: "2 Jahre",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Google Analytics wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen."
                  },
                  {
                    name: "_gid",
                    domain: "google.com",
                    expiration: "1 Tag",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Google Analytics wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen.",
                  },
                  {
                    name: "_pc_vis",
                    domain: window.location.hostname,
                    expiration: "1 Jahr",
                    description: "Dieses Cookie wird verwendet, um wiederkehrende Nutzer zu identifizieren, um ein personalisiertes Browsing-Erlebnis zu bieten.",
                  },
                  {
                    name: "_lr",
                    domain: window.location.hostname,
                    expiration: "3 Tage",
                    description: "Dieses Cookie wird verwendet, um wiederkehrende Nutzer zu identifizieren, um ein personalisiertes Browsing-Erlebnis zu bieten.",
                  }
                ]
              }
            },
            {
              title: "Werbe- und Targeting-Cookies",
              description: "Diese Cookies sammeln Informationen darüber, wie du die Website nutzt, welche Seiten du besuchst und welche Links du anklickst. Alle Daten sind anonymisiert und können nicht dazu verwendet werden, dich zu identifizieren.",
              linkedCategory: "marketing",
              cookieTable: {
                headers: {
                  name: "Name",
                  domain: "Service",
                  description: "Zweck",
                  expiration: "Ablaufdatum"
                },
                body: [
                  {
                    name: "^_cl oder CLID",
                    domain: "bestellen.deliverone.de",
                    expiration: "1 Woche",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Microsoft Clarity, ein Usability Tool, wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen.",
                  },
                  {
                    name: "MUID",
                    domain: "bing.com",
                    expiration: "1 Jahr",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Microsoft Ads wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen.",
                  },
                  {
                    name: "MUID",
                    domain: "clarity.ms",
                    expiration: "1 Jahr",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Microsoft Ads wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen.",
                  },
                  {
                    name: "SRM_B",
                    domain: "c.bing.com",
                    expiration: "1 Jahr",
                    description: "Enthält eine zufallsgenerierte User-ID. Anhand dieser ID kann Microsoft Ads wiederkehrende User auf dieser Website wiedererkennen und die Daten von früheren Besuchen zusammenführen.",
                  },
                  {
                    name: "MR",
                    domain: "c.clarity.ms",
                    expiration: "1 Woche",
                    description: "Gibt an, ob der Cookie MUID aktualisiert werden muss.",
                  },
                  {
                    name: "SM",
                    domain: "c.clarity.ms",
                    expiration: "Session",
                    description: "Dient zur Synchronisierung der ID über verschiedene Microsoft-Domains hinweg.",
                  },
                  {
                    name: "ANONCHK",
                    domain: "c.clarity.ms",
                    expiration: "1 Tag",
                    description: 'Gibt an, ob der Cookie "MUID" auf "ANID" übertragen wird. Der Microsoft-Dienst "Clarity" setzt dies automatisch auf 0.',
                  },
                  {
                    name: "_clck",
                    domain: window.location.hostname,
                    expiration: "1 Jahr",
                    description: "Sammelt Daten über die Navigation und das Verhalten des Benutzers auf der Website - Daraus werden statistische Berichte und Heatmaps für den Website-Besitzer erstellt.",
                  }
                ]
              }
            },
            {
              title: "Weitere Informationen",
              description: "Wenn du Fragen zu unserer Cookie-Politik und deinen Wahlmöglichkeiten hast, wende dich bitte an an uns. <a class=\"cc__link\" href=\"https://deliverone.de/contact\" target=\"_blank\">Kontakt</a>."
            }
          ]
        }
      }
    }
  },
  disablePageInteraction: true
});
```

**Comment by orestbida** [comment]
@BerkeAras it works just fine on my end, both on desktop and mobile (chrome and firefox). I don't see the modal prompt again.

**Comment by BerkeAras** [comment]
@orestbida You're right. It works for me on mobile, but not on desktop.
Another question: I have the website https://87616mod.de - there is a Cookie Banner too. Is it possible to transfer the acceptance to https://shop.87616mod.de (there is a button), so the user wont have to accept again?

**Comment by orestbida** [comment]
> but not on desktop

Which browser are you referring to? Have you tried other browsers?

> Is it possible to transfer the acceptance to https://shop.87616mod.de/ (there is a button), so the user wont have to accept again?

You can do it, but only if you use the same cookieconsent version on both domains.

**Comment by BerkeAras** [comment]
@orestbida It was Chrome. Safari works well.

How could I do it? Is it anywhere in the docs (of the new version)?

**Comment by orestbida** [comment]
Currently it's not documented. In order for this to work properly you should adopt:
1. same plugin version
2. same config.  

for both domains (main domain and subdomain).

To enable it, you must specify the main domain name in the config. of your subdomain.

```js
cookie: {
    domain: '87616mod.de'
}
```

**Comment by BerkeAras** [comment]
Thanks @orestbida
But what if the 2 domains are different? (87616mod.de has a link to bestellen.deliverone.de)

**Comment by orestbida** [comment]
TLDR; it's not possible (very complicated)

If you have these 2 domains:
*  `a.com`
*  `sub.a.com` 

you can access/set the cookie in the main domain, from the subdomain, easily.

If you have 2 different domains: 
*  `a.com`
* `b.com`

there's no easy way to access and set the cookie on `a.com` from `b.com` (for security reasons). You'd need to have a custom service/api which stores the consent settings for each user (ip address).

Currently there's an experimental feature called [`sharedStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Shared_Storage_API#browser_compatibility), which should technically make cross domain data sharing much easier. At the moment it's supported by chromium browsers only.

**Comment by BerkeAras** [comment]
Okay, thank you, @orestbida 
BTW: Great work!

**Comment by orestbida** [comment]
thank you!
